src/main/java/bio/terra/workspace/service/datareference/DataReferenceService.java

```diff
@@ -87,6 +87,16 @@ public DataReference createDataReference(
 
     workspaceService.validateWorkspaceAndAction(
         userReq, referenceRequest.workspaceId(), SamConstants.SAM_WORKSPACE_WRITE_ACTION);
+    try {
+      dataReferenceDao.getDataReferenceByName(
+          referenceRequest.workspaceId(),
+          referenceRequest.referenceType(),
+          referenceRequest.name());
+      throw new DuplicateDataReferenceException(
```

Comment:
it would be good to include the name and type and workspace iD in the error message.

Reply:
My concern with that is that `name` is a user provided string, and the input from security has been to avoid directly returning user-provided strings where possible. A common pattern we use instead is returning an error message without the values and logging the actual values, which I've added here.

Reply:
Sounds good.

Reply:
And there is nothing dangerous about returning the reference type or workspaceId is there?
So if we fixed all of that, you could skip the logging and just throw and let the global exception handler do the logging.

Reply:
Makes sense, I'll add bucket and dataset name validation and then return the names as part of the error response.